tests/integration/route_test.py

```diff
@@ -143,6 +156,75 @@ def test_add_static_route_without_next_hop_address(eth1_up):
     assert_routes(routes, cur_state)
 
 
+@ip_monitor_assert_stable_link_up("eth1")
+def test_route_next_hop_address_default(eth1_up):
+    libnmstate.apply({Interface.KEY: [ETH1_INTERFACE_STATE]})
+
+    # NM's routes without next-hop, nmstate with it's representation of an
+    # empty next-hop: '0.0.0.0' or '::'.
+    # If nmstate wouldn't consider them equal it would do deactivate/reactivate.
+    # Ensure that we only reapply with ip_monitor_assert_stable_link_up.
+    _nmcli_con_modify_routes_reapply("eth1", IPV4_TEST_NET1, IPV6_TEST_NET1)
```

Comment:
Can you explain why use `nmcli modify` instead apply another nmstate YAML?

I think the original issue report does not use nmcli.

Reply:
Because if I use nmstate YAML it is dependant to the internal implementation of nmstate how to create the NM connection. I have seen that in some cases it is adding table=254, for example, while I want to test the case that the NM profile exists without `table`. So what I do here is to force certain combinations of initial NM's state of NM and nmstate desired states.

Reply:
Please consider use nmstate to reproduce the issue as the reporter did.
Otherwise, move the test case to `tests/integration/nm` folder as nm specific tests which will be dropped when we move away from NM1.

So please consider reproduce this issue use nmstate only.

Reply:
The bug is caused by NM specific code, it cannot be reliably reproduced with nmstate only. Further modifications to nmstate could make that the integration test always pass, even if the bug were reintroduced.
Moved to /nm subfolder.